pennylane/labs/resource_estimation/resources_base.py

```diff
@@ -0,0 +1,384 @@
+
+        return clean_gate_counts
+
+    def __str__(self):
+        """Generates a string representation of the Resources object."""
+        qm = self.qubit_manager
+        total_qubits = qm.total_qubits
+        total_gates = sum(self.clean_gate_counts.values())
+
+        total_gates_str = str(total_gates) if total_gates <= 999 else f"{Decimal(total_gates):.3E}"
+        total_qubits_str = (
+            str(total_qubits) if total_qubits <= 9999 else f"{Decimal(total_qubits):.3E}"
+        )
+
+        items = "--- Resources: ---\n"
+        items += f" Total qubits: {total_qubits_str}\n"
+        items += f" Total gates : {total_gates_str}\n"
+
+        qubit_breakdown_str = f"clean qubits: {qm.clean_qubits}, dirty qubits: {qm.dirty_qubits}, algorithmic qubits: {qm.algo_qubits}"
+        items += f" Qubit breakdown:\n  {qubit_breakdown_str}\n"
+
+        gate_type_str = ", ".join(
+            [
+                f"'{gate_name}': {Decimal(count):.3E}" if count > 999 else f"'{gate_name}': {count}"
+                for gate_name, count in self.clean_gate_counts.items()
+            ]
+        )
+        items += " Gate breakdown:\n  {" + gate_type_str + "}"
+
+        return items
+
+    def __repr__(self):
+        """Compact string representation of the Resources object"""
+        return {
+            "qubit manager": self.qubit_manager,
+            "gate types": self.gate_types,
+        }.__repr__()
+
+    def _ipython_display_(self):  # pragma: no cover
+        """Displays __str__ in ipython instead of __repr__"""
+        print(str(self))
+
+
+def add_in_series(first: Resources, other) -> Resources:  # +
+    r"""Add two resources assuming the circuits are executed in series.
+
+    Args:
+        first (Resources): first resource object to combine
+        other (Resources): other resource object to combine with
+
+    Returns:
+        Resources: combined resources
+    """
+    qm1, qm2 = (first.qubit_manager, other.qubit_manager)
+
+    new_clean = max(qm1.clean_qubits, qm2.clean_qubits)
+    new_dirty = qm1.dirty_qubits + qm2.dirty_qubits
+    new_budget = qm1.tight_budget or qm2.tight_budget
+    new_logic = max(qm1.algo_qubits, qm2.algo_qubits)
+
+    new_qubit_manager = QubitManager(
+        work_wires={"clean": new_clean, "dirty": new_dirty}, tight_budget=new_budget
+    )
+
+    new_qubit_manager.algo_qubits = new_logic
+    new_gate_types = _combine_dict(first.gate_types, other.gate_types)
+    return Resources(new_qubit_manager, new_gate_types)
+
+
+def add_in_parallel(first: Resources, other) -> Resources:  # &
+    r"""Add two resources assuming the circuits are executed in parallel.
+
+    Args:
+        first (Resources): first resource object to combine
+        other (Resources): other resource object to combine with
+
+    Returns:
+        Resources: combined resources
+    """
+    qm1, qm2 = (first.qubit_manager, other.qubit_manager)
+
+    new_clean = max(qm1.clean_qubits, qm2.clean_qubits)
```

Comment:
If we have a `tight_budget` wouldn't this make us lose total qubits? 🤔

Reply:
A tight_budget doesn't really make sense when adding resources together? For example, if you add two resources in parallel with a tight budget, you will always go over the number of qubits. You do bring up a valid point, it seems like the budget should almost be separate object, independent of the resources. 

I think that's out of the scope of this PR though.